libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,324 @@
+
+["source","sh",subs="attributes"]
 ------------------------------------------------------------------------------
-metricbeat-6.0.0-2017-08-31
+metricbeat-{stack-version}-{localdate}
 ------------------------------------------------------------------------------
 
-And the corresponding Elasticsearch template is named `metricbeat-6.0.0`.
+And the corresponding {es} template is named +metricbeat-{stack-version}+.
 
 This means that each version of the Beat creates a new index, and it is
-guaranteed that the correct template for that version is applied.  With these
+guaranteed that the correct template for that version is applied. With these
 changes in place, you generally don't have to do anything to upgrade the mapping
 template when you move to a new version.
 
-One exception is when upgrading from 5.x, in which case you should make sure to
-read the <<upgrading-to-5.6>> section.
+// REVIEWERS: I'm not sure what the following paragraph means. Why would this
+// result in downtime?
 
 If downtime is not acceptable, another possible approach is to configure a
 different index pattern in the new Beat version, but this will likely require
-adjustments to your Kibana dashboards.
+adjustments to your {kib} dashboards.
+
+[[troubleshooting-upgrade]]
+=== Troubleshooting {beats} upgrade issues
+
+This section describes common problems you might encounter when upgrading to
+{beats} 7.x.
+
+You can avoid some of these problems by reading <<upgrading-6-to-7>> before
+upgrading {beats}.
+
+// REVIEWERS: I really need help fleshing out this section. This is all based
+// on speculation. Not sure I'm describing the symptoms correctly because this
+// content is based on input from reviewers, not my own testing.
+
+[[missing-fields]]
+==== Missing fields or too many fields in the index
+
+You may have run the Beat before loading the required index template. To clean
+up and start again:
+
+. Delete the index that was created when you ran the Beat. For example:
++
+["source","sh",subs="attributes"]
+----
+DELETE metricbeat-{stack-version}-{localdate}*
+----
+
+. If you want the index to work with 6.x dashboards, turn on the compatibility
+layer. See <<enable-ecs-compatibility>>.
+
+. Load the correct index template:
+
+.. If you're sending events directly to {es}, configure the following settings
+in the {beats} configuration file to overwrite the index template:
++
+[source,yaml]
+----
+setup.template.enabled: true
+setup.template.overwrite: true
+----
+// CONSOLE
+
+.. If you're sending events to a different output, such as {ls}, load the
+template manually. See <<load-index-template-manually>>.
+
+. Restart {beats}.
+
+// REVIEWERS: If users mess up and run Filebeat before setting up the index
+// template correctly (for example, if they don't turn on compatibility), is there
+// some way to recover the previous cursor location to avoid dropped or
+// duplicate events?
+
+[[old-dashboards-failing]]
+==== 6.x dashboards not showing data from 7.0 shippers
```

Comment:
Reviewers: Is it expected that users will be able to upgrade their 6.x dashboards (assuming they've customized the dashboards) so that they work with 7.0 shippers? I ran into problems when I tried to do this, so I think there is something missing here.

Reply:
My [test process](https://github.com/elastic/beats/issues/11385) hinged upon preserving 6.5.4 sample dashboards throughout the 6.5.4->6.7->7.0 upgrade process so I believe the answer is yes. What problems did you run into? I might have run into the same ones.

Reply:
@cjcenizal Could very well be user error because I did the fastest/dirtiest upgrade of ES/Kib possible. Here is the error (RC1). I plan to blow this away and install RC2 today. 

```
Error in visualization
Request to Elasticsearch failed: {"error":{"root_cause":[{"type":"query_shard_exception","reason":"failed to create query: {\n \"bool\" : {\n \"must\" : [\n {\n \"query_string\" : {\n \"query\" : \"*\",\n \"default_field\" : \"*\",\n \"fields\" : [ ],\n \"type\" : \"best_fields\",\n \"default_operator\" : \"or\",\n \"max_determinized_states\" : 10000,\n \"enable_position_increments\" : true,\n \"fuzziness\" : \"AUTO\",\n \"fuzzy_prefix_length\" : 0,\n \"fuzzy_max_expansions\" : 50,\n \"phrase_slop\" : 0,\n \"analyze_wildcard\" : true,\n \"escape\" : false,\n \"auto_generate_synonyms_phrase_query\" : true,\n \"fuzzy_transpositions\" : true,\n \"boost\" : 1.0\n }\n },\n {\n \"query_string\" : {\n \"query\" : \"*\",\n \"fields\" : [ ],\n \"type\" : \"best_fields\",\n \"default_operator\" : \"or\",\n \"max_determinized_states\" : 10000,\n \"enable_position_increments\" : true,\n \"fuzziness\" : \"AUTO\",\n \"fuzzy_prefix_length\" : 0,\n \"fuzzy_max_expansions\" : 50,\n \"phrase_slop\" : 0,\n \"analyze_wildcard\" : true,\n \"escape\" : false,\n \"auto_generate_synonyms_phrase_query\" : true,\n \"fuzzy_transpositions\" : true,\n \"boost\" : 1.0\n }\n },\n {\n \"range\" : {\n \"@timestamp\" : {\n \"from\" : null,\n \"to\" : null,\n \"include_lower\" : true,\n \"include_upper\" : true,\n \"boost\" : 1.0\n }\n }\n }\n ],\n \"adjust_pure_negative\" : true,\n \"boost\" : 1.0\n }\n}","index_uuid":"yS8fI87PSdqfIHdkOf1IAg","index":"metricbeat-7.0.0-rc1-2019.04.01-000001"}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":"metricbeat-7.0.0-rc1-2019.04.01-000001","node":"t3Qhp9I9Su6FdHWsWs8qQw","reason":{"type":"query_shard_exception","reason":"failed to create query: {\n \"bool\" : {\n \"must\" : [\n {\n \"query_string\" : {\n \"query\" : \"*\",\n \"default_field\" : \"*\",\n \"fields\" : [ ],\n \"type\" : \"best_fields\",\n \"default_operator\" : \"or\",\n \"max_determinized_states\" : 10000,\n \"enable_position_increments\" : true,\n \"fuzziness\" : \"AUTO\",\n \"fuzzy_prefix_length\" : 0,\n \"fuzzy_max_expansions\" : 50,\n \"phrase_slop\" : 0,\n \"analyze_wildcard\" : true,\n \"escape\" : false,\n \"auto_generate_synonyms_phrase_query\" : true,\n \"fuzzy_transpositions\" : true,\n \"boost\" : 1.0\n }\n },\n {\n \"query_string\" : {\n \"query\" : \"*\",\n \"fields\" : [ ],\n \"type\" : \"best_fields\",\n \"default_operator\" : \"or\",\n \"max_determinized_states\" : 10000,\n \"enable_position_increments\" : true,\n \"fuzziness\" : \"AUTO\",\n \"fuzzy_prefix_length\" : 0,\n \"fuzzy_max_expansions\" : 50,\n \"phrase_slop\" : 0,\n \"analyze_wildcard\" : true,\n \"escape\" : false,\n \"auto_generate_synonyms_phrase_query\" : true,\n \"fuzzy_transpositions\" : true,\n \"boost\" : 1.0\n }\n },\n {\n \"range\" : {\n \"@timestamp\" : {\n \"from\" : null,\n \"to\" : null,\n \"include_lower\" : true,\n \"include_upper\" : true,\n \"boost\" : 1.0\n }\n }\n }\n ],\n \"adjust_pure_negative\" : true,\n \"boost\" : 1.0\n }\n}","index_uuid":"yS8fI87PSdqfIHdkOf1IAg","index":"metricbeat-7.0.0-rc1-2019.04.01-000001","caused_by":{"type":"illegal_argument_exception","reason":"field expansion matches too many fields, limit: 1024, got: 2251"}}}]},"status":400}
```

Reply:
@cjcenizal Maybe I'm not doing all the upgrade steps? Can you describe the process that you followed to upgrade the dashboards? I might be making too many assumptions about what happens when you save a dashboards under a different name.

Reply:
Looks like you're running into the same problem I reported in my issue, too:

```
"field expansion matches too many fields,
 limit: 1024, got: 2251"
```

If you encounter this during the rolling upgrade process, it's due to a bug in Kibana which the team is working on right now.

To upgrade the dashboards, all I did was fix any issues reported by the Upgrade Assistant in Kibana, and then I set `migration.6_to_7.enabled: true` in the Metricbeat 7.0.0 config.